src/ln/router.rs

```diff
@@ -258,29 +258,29 @@ impl RoutingMessageHandler for Router {
 			None => return Err(HandleError{err: "Couldn't find channel for update", action: Some(ErrorAction::IgnoreError)}),
 			Some(channel) => {
 				macro_rules! maybe_update_channel_info {
-					( $target: expr) => {
-						if $target.last_update >= msg.contents.timestamp {
+					( $first_side: expr, $other_side: expr) => {
+						if $first_side.last_update >= msg.contents.timestamp {
 							return Err(HandleError{err: "Update older than last processed update", action: Some(ErrorAction::IgnoreError)});
 						}
-						chan_was_enabled = $target.enabled;
-						$target.last_update = msg.contents.timestamp;
-						$target.enabled = chan_enabled;
-						$target.cltv_expiry_delta = msg.contents.cltv_expiry_delta;
-						$target.htlc_minimum_msat = msg.contents.htlc_minimum_msat;
-						$target.fee_base_msat = msg.contents.fee_base_msat;
-						$target.fee_proportional_millionths = msg.contents.fee_proportional_millionths;
+						chan_was_enabled = $first_side.enabled;
```

Comment:
Nice debugging, but sadly I think you came to the wrong conclusion. The way I read the second paragraph in https://github.com/lightningnetwork/lightning-rfc/blob/master/07-routing-gossip.md#the-channel_update-message reads to me like the original code here is correct and the issue is in the new check - fees/cltv_deltas are announced on the basis of the channel where the HTLC will be forwarded to, not the basis of which channel the HTLC came in on.

Reply:
Yes, fees/cltv_deltas are based on the receiving node policy where the HTLC will be forwarded to, the issue is on the way handle_channel_update is recording the channel_update as one_to_two or two_to_one. When the node is one, and uses the one_to_two side, it has to use the two_to_one directional info and so the receiving node policy. That's this policy in ChannelInfo, i think, is inverted now.

Reply:
The information going into the channels map here looks right to me - its setting the fees on a channel on the entry that represents the outgoing hop (as indicated by it being the side with the src_node_id being equal to the node which signed the message). In order to be calculating the fee on the basis of where the payment is being forwarded to (ie if you route A -> B -> C, the fees paid are the ones that B announced for the B -> C hop) you'd need to change the check that was added to update_add_htlc.